renderer.go

```diff
@@ -235,3 +235,13 @@ func (r *Renderer) print(text string) {
 func (r *Renderer) clearEOL() {
 	fmt.Print(r.formats.clearOEL)
 }
+
+func (r *Renderer) saveCursorPosition() {
+	// todo: move to r.formats and handle zsh, bash, pwsh
+	fmt.Printf("\x1b7")
```

Comment:
I tried this before and somehow this didn't work in some shells. Would be cool to see this actually working as initially this is what I intended to create using the right alignment.

Reply:
Yeah I see what you mean. Things are acting pretty strange for me in bash and zsh. :-/

Reply:
Yup, that's what I had as well.

Reply:
@TravisTX you do need to escape them properly. It might as well have been because of that (I only learned the escaping after trying this).

Reply:
I'm pretty sure I'm escaping correctly. I followed your pattern. Just to be sure on that, I've pushed my changes adjusted for bash and zsh. I have not had much luck getting those shells to cooperate.

Bash: I get some strange behavior when the user's input attempts to word-wrap. Here's a gif of what I'm seeing in bash: https://i.imgur.com/3WmaFxq.gif

Zsh: I also get the strange word-wrap behavior, but additionally the right-aligned segment doesn't display for me.

Have you seen that word-wrap behavior before? Any ideas on that?

Reply:
I tried this as well and I can't get it to work either on zsh/bash. Just like I experienced before. There's however something I'm going to try for zsh specifically.

Reply:
@TravisTX I made #247 which works on ZSH. Maybe we can do something similar for Powershell specifically. As in, the normal behaviour is having blocks which can be moved around. But you have the ability to have 1 block with will be displayed right (as the last one) for certain shells where this actually works.

Reply:
I see that's been merged. I'll change this PR to support `rprompt`. Once rprompt is handled, do we still need the alignment property? We may end up with some weird conflicts if the app supports both.

Reply:
I've been wondering the same. I thinking they can coexist as the purpose of rprompt is to be rendered AFTER the cursor whereas the other one isn't. And some shells do not support this properly (or we need to figure out how).

Reply:
Ready for review 😊